### PR TITLE
[Docs] Fix script-fields snippet execution

### PR DESCRIPTION
**Pull request by cbuescher**
Currently the first snippet in the documentation test in script-fields.asciidoc
isn't executed, although it has the CONSOLE annotation. Adding a test setup
annotation to it seems to fix the problem.

Relates to #27618

### CODE REVIEW

**Comment by cbuescher** [comment]
@nik9000 assigning to you because this comes with some questions to you from my side. This is a follow up to #27618. While the documentation issue reported there is fixed already, I was trying if the test gets executed currently by changing it back to the old bad syntax. Turns out when running `gradle :docs:check -Dtests.method="*script-fields*"` everything passes, while I was expecting an error.
I added the above test setup (using "sales" because I need a numeric field"). I first removed the test setup for the latter snippet on this page, but then those snippets don't get executed. I was under the impression that a singel TEST(setup:...) per page was enough, and that only adding the CONSOLE annotation would also work. Did this change or am I wrong on these points?

**Comment by nik9000** [comment]
> I was under the impression that a singel TEST(setup:...) per page was enough, and that only adding the CONSOLE annotation would also work. Did this change or am I wrong on these points?

A single `// TESTSETUP` is enough but that is different from `// TEST[setup:foo]`. I didn't realize they there look/sound so similar until now. Ooops.

I'll investigate why the snippet isn't running. One moment.

File: docs/reference/search/request/script-fields.asciidoc [hunk 1]
```diff
@@ -31,7 +31,8 @@ GET /_search
 }
 --------------------------------------------------
 // CONSOLE
-
+// TEST[setup:sales]
+// TEST[s/my_field_name/price/]
```
Comment (nik9000) [inline docs/reference/search/request/script-fields.asciidoc]: I'd prefer to change the example so the test lines up a little closer. it isn't a big deal though.

**Review by nik9000 — APPROVED** [review]
It looks to me like we did run the snippet.

```
./gradlew -p docs clean check -Dtests.method="*script-fields*"
grep line_8 docs/build/testrun/integTestRunner/temp/junit4-J0-20180518_170525_24915*.events
```

shows things, it just doesn't fail like you think it ought to.  I *expect* it passes because we don't find any matches at all because the cluster is empty. So the script is never executed.

But this fix is nice because now we actually execute the script and get a much better test.

**Pull request merged**